src/producer.rs

```diff
@@ -422,27 +423,7 @@ impl<Exe: Executor> TopicProducer<Exe> {
 
         let topic = topic.clone();
         let batch_size = options.batch_size;
-        let compression = options.compression;
-
-        match compression {
```

Comment:
Now the compression modes are structs under control. We exclude the corresponding enum members if features are unset. Thus, we're sure that there's no config/implementation mismatch.